CI/missing_include_check.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+RET=0
+ERRORS=0
+
+FILES=$(find  Core/include/ -name "*.hpp" | grep -v "/detail/")
+N_FILES=$(echo "$FILES" | wc -l)
+echo "Check $N_FILES files"
+
+ITER=0
+
+for file in $(find  Core/include/ -name "*.hpp" | grep -v "/detail/"); do
+    ITER=$((ITER+1))
+    echo "$(date +%H:%M:%S)    $((100*ITER/N_FILES))%   check $file"
+    out=$(printf "#include <${file:13}>\nint main() { return 0; }" | clang++ -std=c++17 -O0 -c -I "Core/include" -x c++ - 2>&1)
```

Comment:
Why clang? The static `-I` means it only works with Core headers (and you `find` only those above) right? I'd have thought IWYU should have caused these? Do we have any idea why not?

Reply:
* I think clang compiles a bit faster... but probably it doesn't matter a lot
* Yeah I thought to restrict this to Core at the moment (and just the public headers). I think if one includes any header of the `Core` library (actually, of any library), one shouldn't get a compile error.
* I'm not sure what IWYU actually does. But it does not seem to check if a header is kind of self-consistent with its includes...

Reply:
IWYU should be able to do this but I disabled fixes for the FPE monitoring because it was messing with the OS/compiler includes.

The other problem with IWYU is that we cannot run it on every branch as it takes like 3h on github to complete and it is still kind of noisy and needs manually going through the changes.

Reply:
But there are quite some missing includes in `Core` as well, that IWYU didn't catch...

Reply:
I think it would be a good way to run a check like this here on every PR, and then regularly IWYU to clean up superfluous includes